enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,405 @@
+not be sufficient to specify filtering on multiple architectures. To this end, it would have to support the nodeAffinity field which uses the
+nodeselectorterms to filter on multiple values.
+
+Instead of having OLM dictate the operator deployment on specific nodes, the individual operators themselves would specify which architectures
+they support through a `pod.spec.affinity.nodeAffinity` object which OLM will then provide the ability for the user to override via the Subscription object
+if necessary. This information should also be added to the operands and potentially be made configurable by the user requesting the operand.
+
+Essentially, when operator authors are declaring support for multiple architectures, that entails:
+
+- Making a declaration of that support via the CSV architecture annotation
+- Setting node affinity in the CSV deployment spec so that the operator is scheduled correctly.
+- Update the operator code such that when it creates/updates operand workloads, it sets those same nodeAffinity settings that are set on
+  the operator deployment.
+
+### Machine Config Operator
+
+No changes are needed in the MCO for Phase 1 as the machine-os-content image would be a manifest listed image and the machine config daemon would
+extract the relevant architecture's machine-os-content based on the node it runs on. In the future, architecture specific configurations should
+not be required, instead they need to be templatized and be generalized for certain items, for example kubelet system reserved memory could scale
+based on page size.
+
+### Machine API
+
+- As of today, machinesets can be mutated to change the instance type and image id and this could potentially result in inconsistency between existing
+machines and newly created machines. If the instance type and image is changed to one of a different architecture, there would be workers of
+different architectures with the same machineset definition. While this may not be a problem, it doesn't make for a good user experience. Hence,
+machineset specs will not allowed to be mutated. Anytime a change is required, a new machineset needs to be created and the old one needs to be scaled
+down.
+- Changes are needed in the autoscaler to make it aware of the node architecture when autoscaling from zero. The `kubernetes-io-arch` label needs to be
+set when building the NodeInfo object so the autoscaler can pick the correct pool to scale up for allocating nodes of a particular architecture.
+
+### Hypershift
+
+The addition of heterogeneous support in Openshift also opens up a new dimension for [support][OCPPLAN-5684] in Hypershift where the control plane 
+and the workers could run on different architectures. There would be certain changes needed to realize this as there are limitations which need to
+be addressed:
+
+- When creating the hosted cluster, the latest x86 release payload is picked by default. This would have to change to pick a heterogeneous payload
```

Comment:
There's no default. What defines the payload in Hypershift is the hostedCluster.spec.release.image which is a required field. We have a cli that populates that conveniently, but that's not a contract, the API is.

Reply:
ok yeah my wording wasn't that great..what i meant was ...hypershift should grow support without the user having to change the spec